projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-keyBoardNav.spec.ts

```diff
@@ -0,0 +1,893 @@
+        });
+
+        it('should navigate with arrow Up and Down keys on treeCells', async () => {
+            await testNavigationUpDown(fix, treeGrid, 'ID');
+        });
+
+        it('should navigate with arrow Left and Right keys on parent row', async () => {
+            const columns = ['ID', 'Name', 'HireDate', 'Age', 'OnPTO'];
+            const rowIndex = 0;
+            await testNavigationLeftRight(fix, treeGrid, rowIndex, columns);
+        });
+
+        it('should navigate with arrow Left and Right keys on child row', async () => {
+            const columns = ['ID', 'Name', 'HireDate', 'Age', 'OnPTO'];
+            const rowIndex = 1;
+            await testNavigationLeftRight(fix, treeGrid, rowIndex, columns);
+        });
+
+        it('should move to the top/bottom cell when navigate with Ctrl + arrow Up/Down keys on gridCells', async () => {
+            await testNavigationTopBottom(fix, treeGrid, 'Name');
+        });
+
+        it('should move to the top/bottom cell when navigate with Ctrl + arrow Up/Down keys on treeCells', async () => {
+            await testNavigationTopBottom(fix, treeGrid, 'ID');
+        });
+
+        it('should move to the leftmost/rightmost cell when navigate with Ctrl + arrow Left/Right keys on parent row', async () => {
+            const columns = ['ID', 'Name', 'HireDate', 'Age', 'OnPTO'];
+            const rowIndex = 0;
+            await testNavigationLeftmostRightmostCell(fix, treeGrid, rowIndex, columns);
+        });
+
+        it('should move to the leftmost/rightmost cell when navigate with Ctrl + arrow Left/Right keys on child row', async () => {
+            const columns = ['ID', 'Name', 'HireDate', 'Age', 'OnPTO'];
+            const rowIndex = 2;
+            await testNavigationLeftmostRightmostCell(fix, treeGrid, rowIndex, columns);
+        });
+
+        it('should move to the top left/bottom right cell when navigate with Ctrl + Home/End keys', async () => {
+            const columns = ['ID', 'Name', 'HireDate', 'Age', 'OnPTO'];
+            await testNavigationHomeEnd(fix, treeGrid, columns);
+        });
+
+        it('should move selection when Tab key is pressed ', async () => {
+            const columns = ['ID', 'Name', 'HireDate', 'Age', 'OnPTO'];
+            await testNavigationTab(fix, treeGrid, columns);
+        });
+
+        it('should move selection when Shift + Tab keys are pressed ', async () => {
+            const columns = ['ID', 'Name', 'HireDate', 'Age', 'OnPTO'];
+            await testNavigationShiftTab(fix, treeGrid, columns);
+        });
+
+        it('should expand/callapce row when Alt + arrow Left/Right keys are pressed on a treeCell', async () => {
+            const cellColumn = 'ID';
+            const cellRowIndex = 0;
+            const rowsCount = 10;
+            const rowsCountAfterCollapse = 4;
+            await testExpandCollapse(fix, treeGrid, cellRowIndex, cellColumn, rowsCount, rowsCountAfterCollapse);
```

Comment:
Maybe those constants are not necessary - the values can be fed to the function directly.